### PR TITLE
Directory structure changed

### PR DESCRIPTION
**Pull request by pseusys**
Source directory structure changed, I guess examples and tests directory structures should be changed as well!

### CODE REVIEW

File: dff/context_storages/__init__.py [hunk 1]
```diff
@@ -3,11 +3,18 @@
...
-from .mongo_connector import MongoConnector  # noqa: F401
-from .protocol import PROTOCOLS
+from .sql_connector import (  # noqa: F401
+    SQLConnector,  # noqa: F401
+    postgres_available,  # noqa: F401
+    mysql_available,  # noqa: F401
+    sqlite_available,  # noqa: F401
+    sqlalchemy_available,  # noqa: F401
+)  # noqa: F401
+from .ydb_connector import YDBConnector, ydb_available  # noqa: F401
```
Comment (kudep) [inline dff/context_storages/__init__.py:13]: After #46 merging:
- [ ] mv 'redis_connector' to 'redis' and so on
- [ ] rename RedisConnector to RedisContextStorage

Comment (pseusys) [reply dff/context_storages/__init__.py:13]: Structure of tests was already updated, wasn't it?

Comment (kudep) [reply dff/context_storages/__init__.py:13]: Sorry, I reviewer from a phone and missed it

File: docs/source/conf.py [hunk 2]
```diff
@@ -117,9 +117,9 @@ def setup(_):
     )
     regenerate_apiref(
         [
-            ("dff.connectors.db", "db_connectors"),
-            ("dff.connectors.messenger", "messenger_interfaces"),
-            ("dff.core.engine", "engine"),
-            ("dff.core.pipeline", "pipeline"),
+            ("dff.context_storages", "context_storages"),
+            ("dff.messengers", "messenger_interfaces"),
+            ("dff.script.core", "engine"),
```
Comment (kudep) [inline docs/source/conf.py:122]: `engine` does not exist now

File: tests/script/core/test_labels.py [hunk 3]
```diff
@@ -1,5 +1,5 @@
-from dff.core.engine.core import Context, Actor
-from dff.core.engine.labels import forward, repeat, previous, to_fallback, to_start, backward
+from dff.script import Context, Actor
+from dff.script.labels import forward, repeat, previous, to_fallback, to_start, backward
```
Comment (kudep) [inline tests/script/core/test_labels.py:2]: tests for labels, conds, resps shouldn't be into core

File: tests/connectors/test_dbs.py [hunk 4]
```diff
@@ -3,26 +3,33 @@
...
-from dff.connectors.db.sql_connector import SQLConnector, postgres_available, mysql_available, sqlite_available
-from dff.connectors.db.redis_connector import RedisConnector, redis_available
-from dff.connectors.db.mongo_connector import MongoConnector, mongo_available
-from dff.connectors.db.ydb_connector import YDBConnector, ydb_available
-from dff.connectors.db import connector_factory
+from dff.context_storages import (
+    connector_factory,
+    DBContextStorage,
+    get_protocol_install_suggestion,
+    JSONContextStorage,
```
Comment (kudep) [inline tests/connectors/test_dbs.py:10]: db should to be into `context_storages` for examples and for tests

**Review by kudep — CHANGES_REQUESTED** [review]
- [ ] we have no `connectors` at all
- [ ] remove engine mentioning
- [ ] refactor `generics/responses.py`to `generics.py`
- [ ] change dirs and their content for examples and test
- [ ] follow this structure for all roots
 image
- [ ] need to find and replace many places (in code and documentation) that has phrases like `engine`, `dff.core`, `db_connector` and etc.

File: docs/source/conf.py (hunk 2 above)
Comment (pseusys) [reply docs/source/conf.py:122]: It's just an example name for the documentation section. How should we call it instead?

Comment (pseusys) [reply docs/source/conf.py]: Renamed to `script`

**Review by kudep — CHANGES_REQUESTED** [review]
- [ ] remove `__init__.py` from excluded, instead that use spetial directives into each of all __init__.py files to ignore errors of import
- [ ] replace all rests of old struct, search words `connector.*`

File: tests/pipeline/test_messenger_interface.py [hunk 5]
```diff
@@ -24,7 +25,7 @@
 }
 
 pipeline = Pipeline.from_script(
-    SCRIPT,  # Actor script object, defined in `.utils` module.
+    SCRIPT,  # Actor script object, defined in `.turn_caching` module.
```
Comment (RLKRo) [inline tests/pipeline/test_messenger_interface.py:28]: No.
```suggestion
    SCRIPT,
```

File: tests/pipeline/test_pipeline.py [hunk 6]
```diff
@@ -2,13 +2,13 @@
...
 import sys
 
-import tests.utils as utils
+from tests.test_utils import get_path_from_tests_to_current_dir
 
 
-# TODO: remove this as soon as utils will be moved to PYPI
-sys.path.append(os.path.abspath(f"examples/{utils.get_path_from_tests_to_current_dir(__file__)}"))
+# TODO: remove this as soon as turn_caching will be moved to PYPI
+sys.path.append(os.path.abspath(f"examples/{get_path_from_tests_to_current_dir(__file__)}"))
```
Comment (RLKRo) [inline tests/pipeline/test_pipeline.py:9]: Remove this.
```suggestion
```

**Review by RLKRo — CHANGES_REQUESTED** [review]
- [ ] Rename the branch to `refactor/directory-structure`. This branch doesn't fix any bugs.
- [ ] Modify the description of the PR to fit the [PULL_REQUEST_TEMPLATE](https://github.com/deeppavlov/dialog_flow_framework/blob/dev/.github/PULL_REQUEST_TEMPLATE.md).

File: examples/script/responses/1_basics.py [hunk 7]
```diff
@@ -68,20 +66,17 @@ class CallbackRequest(NamedTuple):
 
 
 def process_request(ctx: Context):
```
Comment (ruthenian8) [inline examples/script/responses/1_basics.py:68]: Repeated definition. Maybe define once in testing_utils?

File: examples/script/responses/2_buttons.py [hunk 8]
```diff
@@ -139,22 +141,15 @@ class CallbackRequest(NamedTuple):
 
 
 def process_request(ctx: Context):
```
Comment (ruthenian8) [inline examples/script/responses/2_buttons.py:143]: Repeated definition. Maybe, define once in testing_utils?

File: examples/script/responses/3_media.py [hunk 9]
```diff
@@ -96,22 +98,17 @@ class CallbackRequest(NamedTuple):
 
 
 def process_request(ctx: Context):
```
Comment (ruthenian8) [inline examples/script/responses/3_media.py:100]: Repeated definition. Maybe, define once in testing_utils?

File: examples/script/responses/1_basics.py (hunk 7 above)
Comment (kudep) [reply examples/script/responses/1_basics.py:68]: I think, we should keep it, maybe in future with best practices, but now it seems good examples of using

**Review by RLKRo — APPROVED** [review]

**Pull request merged**